src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -1,37 +1,51 @@
 package com.mixpanel.android.mpmetrics;
 
 import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
 import android.app.PendingIntent;
+import android.content.Context;
 import android.content.Intent;
 import android.graphics.Bitmap;
 import android.graphics.Color;
 import android.net.Uri;
 import android.test.AndroidTestCase;
 
+import com.mixpanel.android.util.ImageStore;
+
+import junit.framework.Assert;
+
 import org.mockito.ArgumentMatcher;
 
 import static org.mockito.Mockito.*;
 
+import java.time.Instant;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 
 public class MixpanelNotificationBuilderTest extends AndroidTestCase {
 
     private final String VALID_RESOURCE_NAME = "com_mixpanel_android_logo";
     private final int VALID_RESOURCE_ID = R.drawable.com_mixpanel_android_logo;
+    private final String VALID_IMAGE_URL = "https://dev.images.mxpnl.com/1939595/fc767ba09b1a5420bdbcee71c7ae9904.png";
+    private final String INVALID_IMAGE_URL = "http:/badurl";
     private final String INVALID_RESOURCE_NAME = "NOT A VALID RESOURCE";
     private final String DEFAULT_TITLE = "DEFAULT TITLE";
     private final int DEFAULT_ICON_ID = android.R.drawable.sym_def_app_icon;
     private final Intent DEFAULT_INTENT = new Intent(Intent.ACTION_BUG_REPORT); // ACTION_BUG_REPORT is chosen because it's identifiably weird
+    private Context context;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
         System.setProperty("org.mockito.android.target", getContext().getCacheDir().getPath());
 
+        this.context = getContext();
```

Comment:
do we need to do this? doesn't `getContext()` always return the same thing? If we do need to do this, can't we assign it at the same time we declare it (line 37)?

Reply:
I'm actually not sure of the reason but it breaks the testing with a null reference error if done purely as an instance variable